### PR TITLE
Fix errors coming from hpx compute examples

### PR DESCRIPTION
**Pull request by atrantan**
This patch fixes few errors that come from hpx compute examples. The partitioned_vector example still fails at compilation.

### CODE REVIEW

File: examples/compute/cuda/hello_compute.cu [hunk 1]
```diff
@@ -60,7 +60,8 @@ int hpx_main(boost::program_options::variables_map& vm)
             int i = blockDim.x * blockIdx.x + threadIdx.x;
             if(i < N)
                 C[i] = A[i] + B[i];
-        }, d_A.data(), d_B.data(), d_C.data());
+        }, (int*)(d_A.data()), (int*)(d_B.data()), (int*)(d_C.data())
+        );
```
Comment (hkaiser) [inline examples/compute/cuda/hello_compute.cu]: Is this change really necessary? Or is it a left-over?

Comment (hkaiser) [inline examples/compute/cuda/hello_compute.cu]: But this cast is clearly wrong as it happens in host code where `d_A.data()` returns a `target_ptr<int>` and not an `int*`. I rather think this is a bug and should be reported to the clang guys.

Comment (atrantan) [inline examples/compute/cuda/hello_compute.cu]: Casts cancelled in a89f9418bd5218322cfacf5066f532476019f429

Comment (hkaiser) [inline examples/compute/cuda/hello_compute.cu]: The problem I'm referring to is that clang apparently instantiates [this function](https://github.com/STEllAR-GROUP/hpx/blob/master/hpx/compute/cuda/detail/launch.hpp#L67-L71) in host code, which leads to the compilation error you're trying to work around.

Comment (atrantan) [inline examples/compute/cuda/hello_compute.cu]: Is it possible for you to submit a reduced testcase to the Clang guys?

Comment (atrantan) [inline examples/compute/cuda/hello_compute.cu]: Got it from the clang documentation : 

> clang uses merged parsing. This is similar to split compilation, except all of the host and device code is present and must be semantically-correct in both compilation steps.

Comment (hkaiser) [inline examples/compute/cuda/hello_compute.cu]: So they will probably not fix it :/

**Comment by atrantan** [comment]
Now hello_compute example doesn't fail at compilation
